store/store.go

```diff
@@ -33,14 +33,41 @@ const (
 	boltDBFileName = "gaia.db"
 )
 
-// Store represents the access type for store
-type Store struct {
+// BoltStore represents the access type for store
+type BoltStore struct {
 	db *bolt.DB
 }
 
-// NewStore creates a new instance of Store.
-func NewStore() *Store {
-	s := &Store{}
+// GaiaStore is the interface that defines methods needed to store
+// pipeline and user related information.
+type GaiaStore interface {
+	Init() error
+	CreatePipelinePut(createPipeline *gaia.CreatePipeline) error
+	CreatePipelineGet() (listOfPipelines []gaia.CreatePipeline, err error)
+	PipelinePut(pipeline *gaia.Pipeline) error
+	PipelineGet(id int) (pipeline *gaia.Pipeline, err error)
+	PipelineGetByName(name string) (pipline *gaia.Pipeline, err error)
+	PipelineGetRunHighestID(pipeline *gaia.Pipeline) (id int, err error)
+	PipelinePutRun(r *gaia.PipelineRun) error
+	PipelineGetScheduled(limit int) ([]*gaia.PipelineRun, error)
+	PipelineGetRunByPipelineIDAndID(pipelineid int, runid int) (*gaia.PipelineRun, error)
+	PipelineGetAllRuns(pipelineID int) ([]gaia.PipelineRun, error)
+	PipelineGetLatestRun(pipelineID int) (*gaia.PipelineRun, error)
+	PipelineDelete(id int) error
+	UserPut(u *gaia.User, encryptPassword bool) error
+	UserAuth(u *gaia.User, updateLastLogin bool) (*gaia.User, error)
+	UserGet(username string) (*gaia.User, error)
+	UserGetAll() ([]gaia.User, error)
+	UserDelete(u string) error
+}
+
+// Provides an embeddable interface so that not all of the methods need
+// to be overriden in case of a Mock.
+var _ GaiaStore = (*BoltStore)(nil)
```

Comment:
Actually no hahahahaha. :D I thought about this some more and it made no sense that it would influence embedability. This is just compile time interface compliance check for BoltStore. If boltstore would miss a function this line here wouldn't compile. Still neat that partial interface embedding works out of the box.